### PR TITLE
Support bash and zsh completion

### LINKED ISSUE #73: UCP: Support zsh/bash auto-completion for TiUP
## Description

Support zsh/bash auto-completion for TiUP, make the tiup easy to use. We should support component auto-completion if possible.

## Score

* 605

## Mentor(s)

* @lonng

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* Golang

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

### What problem does this PR solve? 

Fixes #73 

### What is changed and how it works?

Add a `tiup completion [bash | zsh]` command. 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  source <(tiup completion bash)
+  ## Write bash completion code to a file and source if from .bash_profile
+  tiup completion bash > ~/.completion.bash.inc
+  printf "
+  # tiup shell completion
+  source '$HOME/.completion.bash.inc'
+  " >> $HOME/.bash_profile
+  source $HOME/.bash_profile
+
+
```
Comment (lonng) [inline cmd/completion.go:49]: Does here need two blank lines? can we just retain one blank line?

File: cmd/completion.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 1 {
+				switch args[0] {
+				case "bash":
+					return cmd.Parent().GenBashCompletion(os.Stdout)
+				case "zsh":
+					return genCompletionZsh(os.Stdout, cmd.Parent())
+				default:
+					return errors.New("Unsupported shell type " + args[0])
```
Comment (lonng) [inline cmd/completion.go:64]: The error should begin with the lower-case according to Golang idiomatic.

**Comment by lonng** [comment]
@yeya24 Awesome works. is there any way for components completion?

e.g:
```
$ tiup <TAB>
cluster playground ....
```

File: cmd/completion.go (hunk 1 above)
Comment (yeya24) [reply cmd/completion.go]: Done

File: cmd/completion.go (hunk 2 above)
Comment (yeya24) [reply cmd/completion.go]: Done

**Comment by yeya24** [comment]
> @yeya24 Awesome works. is there any way for components completion?
> 
> e.g:
> 
> ```
> $ tiup <TAB>
> cluster playground ....
> ```

To be honest I don't know how to support that.

**Comment by yeya24** [comment]
@lonng  Can I ask why the CI test failed?  Is it related to this change?

**Comment by lucklove** [comment]
> Can I ask why the CI test failed? Is it related to this change?

@yeya24 Our global CDN doesn't work very well at this time, so the CI may fail on the network issue (and retry will success). We are trying to fix this.

**Pull request merged**